### ISSUE TITLE
Greedy decoding

### ISSUE THREAD

**Issue by Imartinezcuevas** [Imartinezcuevas/nano-llm#11]
Implement greedy decoding using argmax over logits. This serves as the baseline generation method.
